outdoorsmanship.lic

```diff
@@ -24,30 +27,44 @@ class Outdoorsmanship
     @training_spells = @settings.crafting_training_spells
     @targetxp = 4
     @targetxp = args.mindstates.to_i if args.mindstates
+    @worn_trashcan = @settings.worn_trashcan
+    @worn_trashcan_verb = @settings.worn_trashcan_verb    
+    @outdoors_room = @settings.outdoors_room
+    if args.room
+      room = args.room.split('=')[1]
+      @outdoors_room = room
+    end
+
+    @train_method = args.forage || @settings.outdoors_method
 
     @forage_item = if args.collect_item
                      args.collect_item
-                   else
+                   elsif @settings.forage_item
                      @settings.forage_item
+                   elsif DRSkill.getrank('Outdoorsmanship') < 20
+                     'rock'
                    end
+
     if args.perception
       @skill_name = 'Perception'
     else
       @skill_name = 'Outdoorsmanship'
     end
+
     start_exp = DRSkill.getxp(@skill_name)
-    @end_exp = [start_exp + @targetxp, 29].min
+    @end_exp = [start_exp + @targetxp, 34].min
 
+    DRCT.walk_to(@outdoors_room)
     train_outdoorsmanship
   end
 
   def train_outdoorsmanship
     attempt = 0 # Failsafe counter to prevent an infinite loop if item isn't collectable in the room
     while (DRSkill.getxp(@skill_name) < @end_exp) && (attempt < @targetxp)
-      if DRSkill.getrank('Outdoorsmanship') < 20
+      if DRSkill.getrank('Outdoorsmanship') < 20 || @train_method == 'forage' 
```

Comment:
💡 Thoughts on moving `DRSkill.getrank('Outdoorsmanship') < 20` up to where `@train_method` is being defined so that the skill check is next to the same skill check that decides the `@forage_item` should be rock.

Then here the IF statement would be
```ruby
if @train_method == 'forage'
    ... # forage
else
   ... # collect
end
```